File: Latihan A/PC.cpp
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

class PC{

	private:
		//atribut private
		Processor pObj; //data prosesor
		Disk dObj; //data disk
		Ram rObj; //data ram
		int totalPrice = 0; //total harga pc

	public:

		//konstruktor
		PC(){

		}

		PC(Processor p, Disk d, Ram r, int tp){
			this->pObj = p;
			this->dObj = d;
			this->rObj = r;
			this->totalPrice = tp;
		}

		//getter dan setter atribut pObj
		void setProcessor(Processor p){
			this->pObj = p;
		}

		Processor getProcessor(){
			return this->pObj;
		}

		//getter dan setter atribut dObj
		void setDisk(Disk d){
			this->dObj = d;
		}

		Disk getDisk(){
			return this->dObj;
		}

		//getter dan setter atribut rObj
		void setRam(Ram p){
			this->rObj = p;
		}

		Ram getRam(){
			return this->rObj;
		}

		//getter dan setter atribut totalPrice
		void setTotalPrice(int p1, int p2, int p3){
			this->totalPrice = p1+p2+p3;
		}

		int getTotalPrice(){
			return this->totalPrice;
		}

		void printPC(){

			cout << "-------------------------" << endl;
			cout << "Processor" << endl;
			cout << "----------" << endl;
			cout << "Name     : " << pObj.getName() << endl;
			cout << "Price    : Rp" << pObj.getPrice() << endl;
			cout << "-------------------------" << endl;
			cout << "Disk" << endl;
			cout << "----------" << endl;
			cout << "Type     : " << dObj.getType() << endl;
			cout << "Capacity : " << dObj.getCapacity() << " GB" << endl;
			cout << "Price    : Rp" << dObj.getPrice() << endl;
			cout << "-------------------------" << endl;
			cout << "Ram" << endl;
			cout << "----------" << endl;
			cout << "Capacity : " << rObj.getCapacity() << " GB" << endl;
			cout << "Price    : Rp" << rObj.getPrice() << endl;
			cout << "-------------------------" << endl;

			cout << "Total Price: Rp" << this->totalPrice << endl;

		}

		//destruktor
		~PC(){

		}

};

File: Latihan B/PHP/PC.php
<?php

include "Processor.php";
include "Disk.php";
include "Ram.php";

class PC{

	//atribut private
	private $pObj; //data prosesor
	private $dObj; //data disk
	private $rObj; //data ram
	private $totalPrice; //total harga pc


	//konstruktor
	public function __construct(){
		$this->pObj = new Processor();
		$this->dObj = new Disk();
		$this->rObj = new Ram();
		$this->totalPrice = 0;
	}

	//getter dan setter atribut pObj
	public function setProcessor($pObj){
		$this->pObj = $pObj;
	}

	public function getProcessor(){
		return $this->pObj;
	}

	//getter dan setter atribut dObj
	public function setDisk($dObj){
		$this->dObj = $dObj;
	}

	public function getDisk(){
		return $this->dObj;
	}

	//getter dan setter atribut rObj
	public function setRam($rObj){
		$this->rObj = $rObj;
	}

	public function getRam(){
		return $this->rObj;
	}

	//getter dan setter atribut totalPrice
	public function setTotalPrice($p1, $p2, $p3){
		$this->totalPrice = $p1+$p2+$p3;
	}

	public function getTotalPrice(){
		return $this->totalPrice;
	}

	//destruktor
	public function __destruct(){

	}

}

?>